File: system-test/typescript/src/testkit/dsl/port/myShop/given/steps/given-promotion.ts
import type { GivenStage } from '../given-stage.js';
import type { WhenStage } from '../../when/when-stage.js';
import type { ThenStage } from '../../then/then-stage.js';

export interface GivenPromotion {
  withActive(promotionActive: boolean): GivenPromotion;
  withDiscount(discount: string | number): GivenPromotion;
  and(): GivenStage;
  when(): WhenStage;
  then(): ThenStage;
}

File: system-test/java/src/main/java/com/mycompany/myshop/testkit/dsl/port/given/steps/GivenPromotion.java
package com.mycompany.myshop.testkit.dsl.port.given.steps;

import com.mycompany.myshop.testkit.dsl.port.given.steps.base.GivenStep;

public interface GivenPromotion extends GivenStep {
    GivenPromotion withActive(boolean promotionActive);
    GivenPromotion withDiscount(double discount);
    GivenPromotion withDiscount(String discount);
}
